packages/lit-dev-content/site/articles/article/lit-cheat-sheet.md

```diff
@@ -0,0 +1,1131 @@
+reactive controller's `hostUpdate()` and `hostUpdated()`.
+
+{% playground-ide "articles/lit-cheat-sheet/signal-watcher", true %}
+
+**Related Documentation & Topics:**
+
+- [Signals docs](/docs/data/signals/)
+- [Lit lifecycle](/docs/components/lifecycle/)
+- [`@lit-labs/signals` npm package](https://www.npmjs.com/package/@lit-labs/signals) (external)
+- [Signals standards proposal](https://github.com/tc39/proposal-signals) (external)
+
+### Pinpoint Signal Updates (watch directive)
+
+The `watch()` directive allows you to pinpoint exactly where a signal should
+update the DOM without re-triggering the lit-html re-render. What this means is
+that using the `watch()` directive will not trigger the `render()` unless it
+triggers the change of a traditional Lit Reactive Property.
+
+{% playground-ide "articles/lit-cheat-sheet/signals-watch-directive", true %}
+
+{% aside "warn" "no-header" %}
+
+This may be a helpful way to optimize performance in your Lit components, but
+*always measure for your use case*.
+
+{% endaside %}
+
+**Related Documentation & Topics:**
+
+- [The `watch()` directive](https://www.npmjs.com/package/@lit-labs/signals#watch-directive) (external)
+- [Reactive Properties](#reactive-properties)
+- [Signals docs](/docs/data/signals/)
+- [Lit lifecycle](/docs/components/lifecycle/)
+- [`@lit-labs/signals` npm package](https://www.npmjs.com/package/@lit-labs/signals) (external)
+- [Signals standards proposal](https://github.com/tc39/proposal-signals) (external)
+
+### Signals HTML Template Tag
+
+The `@lit-labs/signals` package also provides an `html` template tag that can
+be used in place of Lit's default `html` tag and automatically wraps any signals
+in the template with a `watch()` directive.
+
+{% playground-ide "articles/lit-cheat-sheet/signals-html-tag", true %}
+
+**Related Documentation & Topics:**
+
+- [The `html` tag and `withWatch()`](https://www.npmjs.com/package/@lit-labs/signals#html-tag-and-withwatch) (external)
+- [the `SignalWatcher()` mixin](#common-signal-setup-(signalwatcher))
+- [The `watch()` directive](#pinpoint-signal-updates-(watch-directive))
+
+### Make Signal Values from Other Signals (computed)
+
+Sometimes you need to derive a value from other signals. You can do this with
+a `computed()` signal.
+
+{% playground-ide "articles/lit-cheat-sheet/signals-computed", true %}
+
+**Related Documentation & Topics:**
+
+- [Signals docs](/docs/data/signals/)
+- [`@lit-labs/signals` npm package](https://www.npmjs.com/package/@lit-labs/signals) (external)
+
+### Reacting to Signal Changes (effects)
+
+The official `signal-utils` package currently provides an experimental
+`effect()` function that allows you to react to signal changes and run side
+effects.
+
+{% playground-ide "articles/lit-cheat-sheet/signals-effect", true %}
+
+{% aside "warn" %}
+
+The `effect()` function from the `signal-utils` package is experimental.
+
+The current implementation as of December 2024 does not have a way to clean up
```

Comment:
I don't think this is the case, see https://github.com/proposal-signals/signal-utils/blob/main/src/subtle/microtask-effect.ts#L38.

You just have to call the returned function yourself. Let's change this to remind users to do this in disconnected. I also changed the example to do this.

Reply:
huh idk why I was running into this issue before